src/argument.cpp

```diff
@@ -23,7 +23,11 @@
  */
 #include <migraphx/argument.hpp>
 #include <migraphx/functional.hpp>
+#include <migraphx/make_shared_array.hpp>
+#include <migraphx/shape.hpp>
+#include <functional>
 #include <unordered_map>
+#include <utility>
```

Comment:
If this is for clang-tidy we should just disable the check for now. I am thinking we might want to disable it completely as this will just add a ton of includes to every file.